src/Aspire.Hosting.Azure/Provisioning/Provisioners/BicepProvisioner.cs

```diff
@@ -162,15 +164,20 @@ await notificationService.PublishUpdateAsync(resource, state =>
 
         resourceLogger.LogInformation("Deploying {Name} to {ResourceGroup}", resource.Name, resourceGroup.Name);
 
-        var deployments = resourceGroup.GetArmDeployments();
+        // Deploy-time provisioning should target the subscription scope while run-time
+        // provisioning should target the resource group scope.
+        var deployments = context.ExecutionContext.IsPublishMode
```

Comment:
This is one of the places where we need to differentiate between run-mode and publish-mode in the provisioner. Deployments during publish-mode are subscription scoped and need to be resolved differently.

Reply:
This shouldn't be based on the mode. This should be part of the Scope property on the AzureBicepResource. We'll need to do this https://github.com/dotnet/aspire/issues/7514 and I suspect it'll look similar to this.